### DISCUSSION TITLE
Train GAT in an unsupervised way?

### DISCUSSION THREAD

**Discussion by ZRH0308** [pyg-team/pytorch_geometric#6002] [Q&A]
Hi!
I need to train GAT in an unsupervised way like GraphSAGE [https://github.com/pyg-team/pytorch_geometric/blob/b57489fd759273a4a7ba4a9613fa87831daa937d/examples/graph_sage_unsup.py](https://github.com/pyg-team/pytorch_geometric/discussions/url), 
is there any problem if I do this?

**Comment by ZRH0308** [comment]
I have tried this and got better results on Cora than GraphSAGE (acc 79% vs. 75%), but I'm not sure if there are any potential risks?

**Reply by EdisonLeeeee** [reply]
Look correct to me. Both GraphSAGE and GAT are inductive methods and they should be equivalent in most cases.

**Reply by ZRH0308** [reply]
> Look correct to me. Both GraphSAGE and GAT are inductive methods and they should be equivalent in most cases.

Ok!  Thank you for your reply!
